src/Microsoft.TemplateEngine.Core/Operations/InlineMarkupConditional.cs

```diff
@@ -169,7 +176,19 @@ private void FindEnd(IProcessorState processorState, ref int bufferLength, ref i
                         int token;
                         if (_structureTrie.GetOperation(processorState.CurrentBuffer, bufferLength, ref currentBufferPosition, out token))
                         {
-                            if (token == _mapping.OpenOpenElementToken)
+                            if (inComment)
+                            {
+                                // When in comment keep skipping everything else until comment closed
+                                if (token == _mapping.CloseCommentToken)
```

Comment:
for the following case
```
<Smth Condition= "a == true" >
  <!-- #ifdef (b == true) -->
     smth
  <!-- #endif>
</Smth>
```

how the inner condition will be processed? it seems that it is just skipped now.

Reply:
To be on a sure side - there is a dedicated unit test for this specific case: https://github.com/dotnet/templating/pull/5110/files#diff-908b1982784eae2b6a5f2cf512da6bea315f95b84d2bd54ac56bd20fa467d654R42

But in short - the `FindEnd` is not executed for positive condition results